### PR TITLE
Front end proposed changes for query display

### PR DESCRIPTION
**Pull request by DavidStables**
Proposed front end changes as in document for a more flexible query display

### CODE REVIEW

**Review by kkardatos — APPROVED** [review]

**Review by VororJo — APPROVED** [review]

File: ui/src/components/query/viewer/RecursiveMatchDisplay.vue [hunk 1]
```diff
@@ -0,0 +1,169 @@
...
+        v-if="isArrayHasLength(match.match)"
+        v-for="(nestedQuery,index) in match.match"
+        :inline="false"
+        :match="nestedQuery"
+        :key="index"
+        :depth="1"
+    />
+    <RecursiveWhereDisplay
+        v-if="isArrayHasLength(match.where)"
+        v-for="(nestedWhere,index) in match.where"
```
Comment (matt-voror) [inline ui/src/components/query/viewer/RecursiveMatchDisplay.vue:23]: if and for

File: ui/src/components/query/viewer/RecursiveMatchDisplay.vue [hunk 2]
```diff
@@ -0,0 +1,169 @@
+<template>
+  <component
+      :is="!inline ? 'div' : 'span'"
+      :style="indentationStyle(inline,depth)">
+    <span v-if="match.includeIf" :class="'then'">{{ match.includeIf }}</span>
```
Comment (matt-voror) [inline ui/src/components/query/viewer/RecursiveMatchDisplay.vue:5]: no binding on string classes

File: ui/src/components/query/viewer/RecursiveMatchDisplay.vue [hunk 3]
```diff
@@ -0,0 +1,169 @@
...
+  isVariable? : boolean;
+  depth: number;
+  inline : boolean;
+
+}
+
+const props = defineProps<Props>();
+const op: Ref<any> = ref();
+const clickedNodeRef: Ref<Match> = ref({} as Match);
+const list: Ref<Match[]> = ref([]);
```
Comment (matt-voror) [inline ui/src/components/query/viewer/RecursiveMatchDisplay.vue:67]: too generic name

File: ui/src/components/query/viewer/RecursiveMatchDisplay.vue [hunk 4]
```diff
@@ -0,0 +1,169 @@
...
+interface Props {
+  match: Match;
+  isVariable? : boolean;
+  depth: number;
+  inline : boolean;
+
+}
+
+const props = defineProps<Props>();
+const op: Ref<any> = ref();
```
Comment (matt-voror) [inline ui/src/components/query/viewer/RecursiveMatchDisplay.vue:65]: naming

File: ui/src/components/query/viewer/RecursiveMatchDisplay.vue [hunk 5]
```diff
@@ -0,0 +1,169 @@
...
+const clickedNodeRef: Ref<Match> = ref({} as Match);
+const list: Ref<Match[]> = ref([]);
+const op1: Ref<any> = ref();
```
Comment (matt-voror) [inline ui/src/components/query/viewer/RecursiveMatchDisplay.vue:68]: naming

File: ui/src/components/query/viewer/RecursiveMatchDisplay.vue [hunk 6]
```diff
@@ -0,0 +1,169 @@
...
+        if (item.qualifier)
+          parts.push("<span>"+item.qualifier+"</span>");
+        if (item.name)
+          parts.push("<span style=\"color : rgb(0,102,102);\">"+item.description+"</span>");
+        return parts.join(' ');
+      })
+      .join('-> ');
+  return path;
+}
+// Watch for changes in the prop and update the local copy accordingly
```
Comment (matt-voror) [inline ui/src/components/query/viewer/RecursiveMatchDisplay.vue:103]: Missing TODO functionality?

File: ui/src/components/query/viewer/RecursiveMatchDisplay.vue [hunk 7]
```diff
@@ -0,0 +1,169 @@
...
+.feature-indent {
+  display: flex;
+  flex-flow: column;
+  margin-left: 1rem;
+  margin-top: 0.1rem;
+  margin-bottom: 0.1rem;
+}
+.then {
+  padding-right: 0.2rem;
+  color : rgb(72,32,92)
```
Comment (matt-voror) [inline ui/src/components/query/viewer/RecursiveMatchDisplay.vue:127]: theme color

File: ui/src/components/query/viewer/RecursiveMatchDisplay.vue [hunk 8]
```diff
@@ -0,0 +1,169 @@
...
+}
+.field {
+  padding-right : 0.2rem;
+}
+
+.variable-field {
+  padding-left : 2rem;
+  color : rgb(180, 48, 190);
```
Comment (matt-voror) [inline ui/src/components/query/viewer/RecursiveMatchDisplay.vue:135]: theme color

File: ui/src/components/query/viewer/RecursiveMatchDisplay.vue [hunk 9]
```diff
@@ -0,0 +1,169 @@
...
+  padding-right : 0.3rem;
+}
+
+
+.return {
+  color: lightseagreen;
```
Comment (matt-voror) [inline ui/src/components/query/viewer/RecursiveMatchDisplay.vue:141]: theme color

File: ui/src/components/query/viewer/RecursiveMatchDisplay.vue [hunk 10]
```diff
@@ -0,0 +1,169 @@
...
+  padding-left: 0.5rem;
+}
+
+.output {
+  color: mediumslateblue;
```
Comment (matt-voror) [inline ui/src/components/query/viewer/RecursiveMatchDisplay.vue:146]: theme color

File: ui/src/components/query/viewer/RecursiveMatchDisplay.vue [hunk 11]
```diff
@@ -0,0 +1,169 @@
...
+}
+</style>
+
+<style>
```
Comment (matt-voror) [inline ui/src/components/query/viewer/RecursiveMatchDisplay.vue:150]: replace global style sheet with :deep styles in scoped section and use theme colors
https://vuejs.org/api/sfc-css-features.html

File: ui/src/components/query/viewer/RecursiveQueryDisplay.vue [hunk 12]
```diff
@@ -1,114 +1,92 @@
...
-    "
-  >
-    <span>
-      <span v-if="parentMatch?.boolMatch && matchIndex" :class="parentMatch?.boolMatch">{{ parentMatch?.boolMatch }}</span>
-      <span v-if="match.name" v-html="match.name"> </span>
-      <span v-else-if="match.description" v-html="match.description"> </span>
-      <span v-else-if="match.displayLabel">{{ match.displayLabel }}</span>
+      :style="{ paddingLeft: '1rem' }">
+    <span v-if="!query.match&&!query.where&&!query.return">
+      <span v-if="query.name" style="padding-left:2rem;" v-html="query.name"></span>
```
Comment (matt-voror) [inline ui/src/components/query/viewer/RecursiveQueryDisplay.vue:5]: incorrect use of v-html

File: ui/src/components/query/viewer/RecursiveMatchDisplay.vue [hunk 13]
```diff
@@ -0,0 +1,169 @@
+<template>
+  <component
+      :is="!inline ? 'div' : 'span'"
+      :style="indentationStyle(inline,depth)">
+    <span v-if="match.includeIf" :class="'then'">{{ match.includeIf }}</span>
+    <span v-if="match.inlineOperator" :class="match.inlineOperator">{{ match.inlineOperator}}</span>
+    <span v-if="match.exclude" :class="'field'">NOT</span>
+    <span v-if="match.orderBy" :class="'field'"v-html="match.orderBy.description"></span>
```
Comment (matt-voror) [inline ui/src/components/query/viewer/RecursiveMatchDisplay.vue:8]: incorrect use of v-html on standard string
v-html is a potential security issue and is used to render secure html strings only.
Use `<pre></pre>` element if you need string formating

File: ui/src/components/query/viewer/RecursiveQueryDisplay.vue [hunk 14]
```diff
@@ -1,114 +1,92 @@
...
-      :parent-match="match"
-      :full-query="fullQuery"
-    />
-    <span v-if="match.orderBy"> <div v-html="match.orderBy.description"></div></span>
-    <span v-if="match.then">
-      <RecursiveQueryDisplay :match="match.then" :parent-match="match" :full-query="fullQuery" />
+
+    <span v-if="isArrayHasLength(query.match)||isArrayHasLength(query.where)||isArrayHasLength(query.return)">
+        <Button class="button-chevron" @click="matchToggle" >
+      <IMFontAwesomeIcon
```
Comment (matt-voror) [inline ui/src/components/query/viewer/RecursiveQueryDisplay.vue:10]: see prev example

File: ui/src/components/query/viewer/RecursiveQueryDisplay.vue [hunk 15]
```diff
@@ -1,114 +1,92 @@
...
+          :icon="!matchExpand ? ['fa-solid','fa-chevron-right'] :  ['fa-solid','fa-chevron-up']"
+          :style="'color: blue'"
+          class="mr-2"
+          fixed-width
+      />
+        </Button>
+       <span v-if="query.name" v-html="query.name"></span>
```
Comment (matt-voror) [inline ui/src/components/query/viewer/RecursiveQueryDisplay.vue:17]: incorrect use of v-html

File: ui/src/components/query/viewer/RecursiveQueryDisplay.vue [hunk 16]
```diff
@@ -1,114 +1,92 @@
...
+        :property-index="index"
+        :key="index"
+        :second-property="index === 1"
+      />
     </span>
-    <span v-if="match.variable">
-      label as
-      <span class="variable">{{ match.variable }}</span>
+    <span v-if="matchExpand&&isArrayHasLength(query.return)">
+      <div style="color: green;padding-left: 2rem">Fields (columns)</div>
```
Comment (matt-voror) [inline ui/src/components/query/viewer/RecursiveQueryDisplay.vue:39]: theme color

File: ui/src/components/query/viewer/RecursiveQueryDisplay.vue [hunk 17]
```diff
@@ -1,114 +1,92 @@
...
+  returnExpanded:boolean;
 }
 
 const props = defineProps<Props>();
-
 const op: Ref<any> = ref();
-const clickedNodeRef: Ref<Where | Match> = ref({} as Where);
-const list: Ref<Node[]> = ref([]);
+const clickedNodeRef: Ref<Match> = ref({} as Match);
+const list: Ref<Match[]> = ref([]);
```
Comment (matt-voror) [inline ui/src/components/query/viewer/RecursiveQueryDisplay.vue:70]: naming

File: ui/src/components/query/viewer/RecursiveReturnDisplay.vue [hunk 18]
```diff
@@ -0,0 +1,75 @@
...
+
+
+<style scoped>
+
+
+.field {
+  padding-right : 0.2rem;
+}
+.return {
+  color: lightseagreen;
```
Comment (matt-voror) [inline ui/src/components/query/viewer/RecursiveReturnDisplay.vue:66]: theme color

File: ui/src/components/query/viewer/RecursiveWhereDisplay.vue [hunk 19]
```diff
@@ -0,0 +1,154 @@
+<template>
+    <span class="property-display">
+      <span v-html="getOperator(property.inlineOperator,propertyIndex,secondProperty)"></span>
```
Comment (matt-voror) [inline ui/src/components/query/viewer/RecursiveWhereDisplay.vue:3]: incorrect use of v-html

File: ui/src/components/query/viewer/RecursiveWhereDisplay.vue [hunk 20]
```diff
@@ -0,0 +1,154 @@
...
+      <span v-if="property.name" :class="'field'" v-html="property.name"></span>
```
Comment (matt-voror) [inline ui/src/components/query/viewer/RecursiveWhereDisplay.vue:4]: incorrect use of v-html

File: ui/src/components/query/viewer/RecursiveWhereDisplay.vue [hunk 21]
```diff
@@ -0,0 +1,154 @@
...
+       <span v-if="property.qualifier" :class="'value-field'" v-html="property.qualifier"></span>
```
Comment (matt-voror) [inline ui/src/components/query/viewer/RecursiveWhereDisplay.vue:5]: incorrect use of v-html

File: ui/src/components/query/viewer/RecursiveWhereDisplay.vue [hunk 22]
```diff
@@ -0,0 +1,154 @@
...
+        <span v-if="property.valueLabel" v-html="getFormattedValue(property)"></span>
```
Comment (matt-voror) [inline ui/src/components/query/viewer/RecursiveWhereDisplay.vue:6]: incorrect use of v-html

File: ui/src/components/query/viewer/RecursiveWhereDisplay.vue [hunk 23]
```diff
@@ -0,0 +1,154 @@
...
+         <span v-if="property.relativeTo">
+           <span v-if="property.relativeTo.qualifier">
+             <span :class="'field'" v-html="property.relativeTo.qualifier"></span>
```
Comment (matt-voror) [inline ui/src/components/query/viewer/RecursiveWhereDisplay.vue:9]: incorrect use of v-html

File: ui/src/components/query/viewer/RecursiveWhereDisplay.vue [hunk 24]
```diff
@@ -0,0 +1,154 @@
...
+           </span>
+           <span :class="'node-ref'" v-html="property.relativeTo.nodeRef"></span>
```
Comment (matt-voror) [inline ui/src/components/query/viewer/RecursiveWhereDisplay.vue:11]: incorrect use of v-html

File: ui/src/components/query/viewer/RecursiveWhereDisplay.vue [hunk 25]
```diff
@@ -0,0 +1,154 @@
...
+         </span>
+         <span v-if="property.nodeRef" :class="'node-ref'" v-html="property.nodeRef"></span>
```
Comment (matt-voror) [inline ui/src/components/query/viewer/RecursiveWhereDisplay.vue:13]: incorrect use of v-html

File: ui/src/components/query/viewer/RecursiveWhereDisplay.vue [hunk 26]
```diff
@@ -0,0 +1,154 @@
...
+
+      <div v-if="isArrayHasLength(property.where)"
+           :style="indentationStyle(depth+1)">
+        <span :class="property.operator"> {{property.operator}}</span>
+          <span>(</span>
+        <RecursiveWhereDisplay
+          v-if="isArrayHasLength(property.where)"
+          v-for="(nestedProperty, index) of property.where"
```
Comment (matt-voror) [inline ui/src/components/query/viewer/RecursiveWhereDisplay.vue:21]: if & for

File: ui/src/components/query/viewer/RecursiveWhereDisplay.vue [hunk 27]
```diff
@@ -0,0 +1,154 @@
...
+  secondProperty: boolean;
+}
+
+const props = defineProps<Props>();
+
+const op: Ref<any> = ref();
+const op1: Ref<any> = ref();
+const childExpand= true;
+const clickedProperty: Ref<Where> = ref({} as Where);
+const list: Ref<Node[]> = ref([]);
```
Comment (matt-voror) [inline ui/src/components/query/viewer/RecursiveWhereDisplay.vue:63]: naming

File: ui/src/components/query/viewer/RecursiveWhereDisplay.vue [hunk 28]
```diff
@@ -0,0 +1,154 @@
...
+interface Props {
+  property: Where;
+  propertyIndex?: any;
+  depth : number;
+  secondProperty: boolean;
+}
+
+const props = defineProps<Props>();
+
+const op: Ref<any> = ref();
```
Comment (matt-voror) [inline ui/src/components/query/viewer/RecursiveWhereDisplay.vue:59]: naming

File: ui/src/components/query/viewer/RecursiveWhereDisplay.vue [hunk 29]
```diff
@@ -0,0 +1,154 @@
...
+const op1: Ref<any> = ref();
```
Comment (matt-voror) [inline ui/src/components/query/viewer/RecursiveWhereDisplay.vue:60]: naming

File: ui/src/components/query/viewer/RecursiveWhereDisplay.vue [hunk 30]
```diff
@@ -0,0 +1,154 @@
...
+const childExpand= true;
+const clickedProperty: Ref<Where> = ref({} as Where);
```
Comment (matt-voror) [inline ui/src/components/query/viewer/RecursiveWhereDisplay.vue:62]: type is Where | undefined. Reformat to use undefined not empty object

File: ui/src/components/query/viewer/RecursiveWhereDisplay.vue [hunk 31]
```diff
@@ -0,0 +1,154 @@
...
+  margin-bottom: 0.1rem;
+}
+
+.field {
+  padding-right : 0.2rem;
+}
+
+
+.value-field {
+  color : rgb(0, 102, 102);
```
Comment (matt-voror) [inline ui/src/components/query/viewer/RecursiveWhereDisplay.vue:123]: theme color

File: ui/src/components/query/viewer/RecursiveWhereDisplay.vue [hunk 32]
```diff
@@ -0,0 +1,154 @@
...
+  padding-right : 0.2rem;
+}
+
+
+.node-field {
+  color : rgb(180, 48, 190);
```
Comment (matt-voror) [inline ui/src/components/query/viewer/RecursiveWhereDisplay.vue:129]: theme color

File: ui/src/components/query/viewer/RecursiveWhereDisplay.vue [hunk 33]
```diff
@@ -0,0 +1,154 @@
...
+  padding-right : 0.3rem;
+}
+
+.variable-line {
+  margin-left: 1rem !important;
+}
+
+.node-ref {
+  color: rgb(138, 67, 138) !important;
```
Comment (matt-voror) [inline ui/src/components/query/viewer/RecursiveWhereDisplay.vue:138]: theme color

File: ui/src/components/query/viewer/RecursiveWhereDisplay.vue [hunk 34]
```diff
@@ -0,0 +1,154 @@
...
+  cursor: pointer !important;
+}
+.or {
+  color: blue;
```
Comment (matt-voror) [inline ui/src/components/query/viewer/RecursiveWhereDisplay.vue:142]: theme color

File: ui/src/components/query/viewer/RecursiveWhereDisplay.vue [hunk 35]
```diff
@@ -0,0 +1,154 @@
...
+  padding-right: 0.3rem;
+}
+
+.and {
+  color: orange;
```
Comment (matt-voror) [inline ui/src/components/query/viewer/RecursiveWhereDisplay.vue:147]: theme color

**Review by matt-voror — CHANGES_REQUESTED** [review]

File: ui/src/components/directory/viewer/QueryDisplay.vue [hunk 36]
```diff
@@ -7,8 +7,39 @@
       </div>
       <div class="query-display">
         <div class="rec-query-display">
-          <div class="include-title" style="color: green">include if</div>
-          <RecursiveQueryDisplay v-if="query" :match="query" :parent-match="undefined" :full-query="query" />
+          <span v-if="query.name" v-html="query.name"> </span>
+          <div v-if="query.typeOf">
+            <span :class="'field'" v-html="query.typeOf.name"></span>
```
[the remark this answers is not in the document]
Comment (DavidStables) [reply ui/src/components/directory/viewer/QueryDisplay.vue]: Thanks. Have changed then to inline css

File: ui/src/components/directory/viewer/QueryDisplay.vue [hunk 37]
```diff
@@ -7,8 +7,39 @@
...
+            <span class="include-title" style="color: green">with the following features</span>
```
[the remark this answers is not in the document]
Comment (DavidStables) [reply ui/src/components/directory/viewer/QueryDisplay.vue:13]: Done (hopefully got them all)

File: ui/src/components/directory/viewer/QueryDisplay.vue [hunk 38]
```diff
@@ -7,8 +7,39 @@
...
+          </div>
+          <MatchSummaryDisplay
+              v-if="isArrayHasLength(query.match)"
+              v-for="(nestedQuery, index) of query.match"
+              :match="nestedQuery"
+              :expanded="false"
+              />
```
[the remark this answers is not in the document]
Comment (DavidStables) [reply ui/src/components/directory/viewer/QueryDisplay.vue:22]: Thanks. have done

File: ui/src/components/directory/viewer/QueryDisplay.vue [hunk 39]
```diff
@@ -7,8 +7,39 @@
...
+          <div v-if="isArrayHasLength(query.query)"  style="padding-left: 2rem">
+            <Button class="button-chevron" @click="toggle" >
+              <IMFontAwesomeIcon
+                  :icon="!dataSetExpanded ? ['fa-solid','fa-chevron-right'] :  ['fa-solid','fa-chevron-up']"
+                  :style="'color: blue'"
```
[the remark this answers is not in the document]
Comment (DavidStables) [reply ui/src/components/directory/viewer/QueryDisplay.vue]: Done, hopefully done the rest

File: ui/src/components/query/viewer/RecursiveWhereDisplay.vue (hunk 33 above)
Comment (DavidStables) [reply ui/src/components/query/viewer/RecursiveWhereDisplay.vue]: ok

File: ui/src/components/query/viewer/RecursiveWhereDisplay.vue (hunk 35 above)
Comment (DavidStables) [reply ui/src/components/query/viewer/RecursiveWhereDisplay.vue:127]: ok

File: ui/src/components/query/viewer/RecursiveWhereDisplay.vue (hunk 32 above)
Comment (DavidStables) [reply ui/src/components/query/viewer/RecursiveWhereDisplay.vue]: ok

File: ui/src/components/query/viewer/RecursiveWhereDisplay.vue (hunk 31 above)
Comment (DavidStables) [reply ui/src/components/query/viewer/RecursiveWhereDisplay.vue]: ok

File: ui/src/services/QueryService.ts [hunk 40]
```diff
@@ -51,6 +51,17 @@ const QueryService = {
     });
   },
 
+  async getDisplayFromQueryIri(iri: string, includeLogicDesc: boolean): Promise<Query> {
+    return axios.get(Env.API + "api/query/public/queryDisplay", { params: { queryIri: iri, includeLogicDesc: includeLogicDesc } });
+  },
+
+  async getDisplayFromQuery(query: Query, includeLogicDesc: boolean): Promise<Query> {
+    return axios.post(Env.API + "api/query/public/queryDisplayFromQuery", query, {
+      params: { includeLogicDesc: includeLogicDesc }
```
Comment (matt-voror) [inline ui/src/services/QueryService.ts:60]: Can't combine POST body and request params. IMAPI will ignore request params. Create a new request model.

**Review by matt-voror — CHANGES_REQUESTED** [review]

**Review by VororRich — APPROVED** [review]

**Review by VororRich — APPROVED** [review]

**Pull request merged**